core/primitives/src/types.rs

```diff
@@ -152,6 +152,9 @@ pub enum StateChangeCause {
     /// State change that happens when we update validator accounts. Not associated with with any
     /// specific transaction or receipt.
     ValidatorAccountsUpdate,
+    /// State change that is happens due to migration that happens in first block of an epoch
+    /// after protocol upgrade
+    Migration { description: String },
```

Comment:
Here, and in `StateChangeCauseView` I worry a bit that we add a free-form string description to a serialized field. This would be the first such string for both enums it seems. Perhaps future proof this a bit with:

```rust
enum StateChangeCauseView {
   Migration { id: MigrationId },
}

#[derive(Serialize,Deserialize)]
enum MigrationId { StorageUsageFix }

impl fmt::Display for MigrationId {}
```

That is, decouple what the migration is from a user-friendly text description.